### PR TITLE
fix #114381 cwd returns to os.homedir() if workspaceFolder does not resolve

### LINKED ISSUE #114381: "Variable ${workspaceFolder} can not be resolved. Please open a folder." when you run a task
- VSCode Version:
- OS Version:

### Steps to Reproduce:

Have an extension with the following
```ts
	const shellExec = new vscode.ShellExecution('ls',
	{
		executable: 'C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe', // or '/bin/bash'
		shellArgs: [ '-c' ]
	});

	const task = new vscode.Task(
		{ type: 'get-help' },
		vscode.TaskScope.Global,
		'Get help online',
		'Test HyperLink',
		shellExec);

        vscode.tasks.executeTask(task);
```

### Expected:

The task executes properly

### Actual:

"Variable ${workspaceFolder} can not be resolved. Please open a folder." Error notification

### Workaround:

If you add:

```
cwd: os.homedir()
```

to the options of `vscode.ShellExecution` it works as expected.

### Proposal

The default value of `cwd` should fall back to `os.homedir()` if `workspaceFolder` does not resolve.

Does this issue occur when all extensions are disabled?: N/A

### PR DESCRIPTION
**Pull request by noviicee**
This PR fixes #114381 

Changes:
`cwd` returns to `os.homedir()` if `workspaceFolder` does not resolve.

### CODE REVIEW

**Comment by noviicee** [comment]
@TylerLeonhardt @alexr00 kindly have a look.

File: src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts [hunk 1]
```diff
@@ -1658,7 +1658,10 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 			try {
 				cwd = await this.resolveVariable(resolver, '${workspaceFolder}');
 			} catch (e) {
-				// No workspace
+				// If we don't have a cwd, then the terminal uses the home dir.
```
Comment (alexr00) [inline src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts:1661]: Let's have a check here for the specific error we're looking for.

**Review by alexr00 — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts [hunk 2]
```diff
@@ -1658,7 +1658,10 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
...
+				let cwd_uri: URI;
+				cwd_uri = await this.pathService.userHome();
+				cwd = cwd_uri.toString();
```
Comment (alexr00) [inline src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts:1664]: Does `toString` have the "file://" prefix? This should maybe be `fspath`.

**Comment by TylerLeonhardt** [comment]
@noviicee just wanna check. Were you able to figure out how to manually verify your change using the test extension I provided in the original issue?

**Comment by noviicee** [comment]
> @noviicee just wanna check. Were you able to figure out how to manually verify your change using the test extension I provided in the original issue?

No I was unable to figure out exactly how to use the test extension and manually verify the change 💭

**Comment by noviicee** [comment]
> Unfortunately, we can't take the change unless you've _at least_ manually tested it. You'll find that a lot of OSS projects will have this requirement so it's a great skill to have. It's pretty easy to test this change:
> 
> * First make sure you can run the OSS version of VS Code: https://github.com/microsoft/vscode/wiki/How-to-Contribute
> * Then you should create a new extension for which there is a guide here: https://code.visualstudio.com/api/get-started/your-first-extension
> 
> > Make sure you open that new extension in OSS otherwise you won't actually be testing your code, would you? :)
> 
> * Grab my code from ["Variable ${workspaceFolder} can not be resolved. Please open a folder." when you run a task  #114381](https://github.com/microsoft/vscode/issues/114381) and put in the `activate` function in `src/extension.ts` in that new extension you made
> 
> > hint: read the new extension guide for how to run your newly made extension
> 
> * Once you have your extension running, you'll want to run the task:
>   
>   * If you need to, run your Hello World command first to activate your extension. ctrl+shift+p/cmd+shift+p on mac and type "Hello World" and hit enter
>   * Then let's run a task: ctrl+shift+p/cmd+shift+p on mac and type "Run Task" and hit enter. Find the right one (it's probably called "Test HyperLink" if you copy and pasted from my code)
>   * At this point you should not see any error

Thanks @TylerLeonhardt for the guidance. Up to it! 🚀

**Comment by alexr00** [comment]
Closing since I have decided to close the original feature request.

**Pull request closed without merging**